src/main/java/com/rybka/Application.java

```diff
@@ -24,16 +31,39 @@ public static void main(String[] args) {
         var exchangeRateConnectorConnector = new ExchangeRateConnector(new CVurl());
 
         var exchangeSourceMap = Map.of(
-                PropertyInfo.PROPERTY_EXCHANGE_KEY, exchangeRateConnectorConnector,
-                PropertyInfo.PROPERTY_PRIME_EXCHANGE_KEY, primeExchangeRateConnector);
+                ExchangeSource.EXCHANGE.getSource(), exchangeRateConnectorConnector,
+                ExchangeSource.PRIME_EXCHANGE.getSource(), primeExchangeRateConnector);
 
         var reader = new PropertyReader(PropertyInfo.PROPERTY_FILE_NAME).getProperties();
-        var connector = exchangeSourceMap.get(reader.getProperty(PropertyInfo.PROPERTY_EXCHANGE_SOURCE));
+        var connector = exchangeSourceMap.entrySet().stream()
+                .filter(item -> reader.getProperty(PropertyInfo.PROPERTY_EXCHANGE_SOURCE).equals(item.getKey()))
+                .findAny()
+                .orElseThrow(() -> new InvalidPropertyException("Cannot find specified source. Try using another one."))
+                .getValue();
+
+        var exportFolder = reader.getProperty(PropertyInfo.PROPERTY_EXPORT_FOLDER);
+        var exportFileName = FileUtils.generateFileName(reader.getProperty(PropertyInfo.PROPERTY_EXPORT_TYPE));
+
+        var consoleExportService = new ConsoleExportService();
+        var csvExportService = new CSVExportService(new CsvMapper(), exportFolder, exportFileName);
+        var jsonExportService = new JSONExportService(objectMapper, exportFolder, exportFileName);
+
+        var exportConfigMap = Map.of(
+                ExportType.CONSOLE.getType(), consoleExportService,
+                ExportType.CSV.getType(), csvExportService,
+                ExportType.JSON.getType(), jsonExportService);
+
+        var exportService = exportConfigMap.entrySet().stream()
```

Comment:
Seems we have almost identical logic for retrieving value from a map. 
Try to create a method which takes <map, someKey> and returns instance or throw an exception in case key is missed in the map.

Reply:
fixed